lib/interface-types/src/interpreter/instructions/strings.rs

```diff
@@ -0,0 +1,386 @@
+                )
+            })?;
+
+            let string: String = to_native(&string, instruction)?;
+            let string_bytes = string.as_bytes();
+            let string_length = string_bytes.len() as i32;
+
+            let outputs = allocator.call(&[InterfaceValue::I32(string_length)]).map_err(|_| {
+                InstructionError::new(
+                    instruction,
+                    InstructionErrorKind::LocalOrImportCall { function_index: allocator_index },
+                )
+            })?;
+            let string_pointer: i32 = to_native(&outputs[0], instruction)?;
+
+            let memory_index: u32 = 0;
+            let memory_view = instance
+                .memory(memory_index as usize)
+                .ok_or_else(|| {
+                    InstructionError::new(
+                        instruction,
+                        InstructionErrorKind::MemoryIsMissing { memory_index }
+                    )
+                })?
+                .view();
+
+            for (nth, byte) in string_bytes.iter().enumerate() {
+                memory_view[string_pointer as usize + nth].set(*byte);
+            }
+
+            runtime.stack.push(InterfaceValue::I32(string_pointer));
+            runtime.stack.push(InterfaceValue::I32(string_length));
+
+            Ok(())
+        }
+    }
+);
+
+executable_instruction!(
+    string_size(instruction: Instruction) -> _ {
+        move |runtime| -> _ {
+            let value = runtime.stack.peek1().ok_or_else(|| {
+                InstructionError::new(
+                    instruction,
+                    InstructionErrorKind::StackIsTooSmall { needed: 1 },
+                )
+            })?;
+
+            if let InterfaceValue::String(string) = value {
+                let length = string.len() as i32;
+                runtime.stack.push(InterfaceValue::I32(length));
+
+                Ok(())
+            } else {
+                Err(InstructionError::new(
+                    instruction,
+                    InstructionErrorKind::InvalidValueOnTheStack {
+                        expected_type: InterfaceType::String,
+                        received_type: value.into(),
+                    }
+                ))
+            }
+        }
+    }
+);
+
+#[cfg(test)]
+mod tests {
+    test_executable_instruction!(
+        test_string_lift_memory =
+            instructions: [
+                Instruction::ArgumentGet { index: 0 },
+                Instruction::ArgumentGet { index: 1 },
+                Instruction::StringLiftMemory,
+            ],
+            invocation_inputs: [
+                InterfaceValue::I32(0),
+                //              ^^^^^^ pointer
+                InterfaceValue::I32(13),
+                //              ^^^^^^^ length
+            ],
+            instance: Instance {
+                memory: Memory::new("Hello, World!".as_bytes().iter().map(|u| Cell::new(*u)).collect()),
```

Comment:
I'm not familiar with `Memory::new` actually, but it seems that we don't need to pass in `Cell` unless this is creating a memory that's the size of the string and using that memory directly.

What I assume is happening is that we're copying these bytes into the real memory so the fact these are wrapped in `Cell` shouldn't matter here as I understand it.  `Cell` is just important from a data creation perspective if that data can be modified without `&mut` but if we're copying it, then that's not a concern.

Reply:
In this case, `Memory` is just a test object, not the real `Memory` struct from `runtime-core` :-).